HPXMLtoOpenStudio/resources/weather.rb

```diff
@@ -341,4 +341,21 @@ def self.calc_mains_temperatures(avgOAT, maxDiffMonthlyAvgOAT, latitude, year)
     end
     return mainsAvgTemp, mainsMonthlyTemps, mainsDailyTemps
   end
+
+  def self.get_undisturbed_ground_temperature(weather, climate_zone_iecc)
```

Comment:
Using TMY3 station data, there are some differences (mainly in cooler climates) between the L. Xing data and the annual drybulb. There is some scatter, but the $r^2$ is pretty good. I would be in favor of using `weather.calc_ground_temperatures()` with an adjustment using the equation below to determine the ground temperature based on the L. Xing Model.

This way we use an established method already implemented in OS-HPXML, only make a small adjustment to the outputs of the method, and ground temperatures change for every location. 

**Drawback:** Different weather files will produce different ground temperatures for a given location (example: an AMY simulation for 2019 and 2020). This might not be bad for foundation heat transfer and shallow GSHP systems but is a little strange for vertical borefields.

![drybulb-ground-temperature-correlation](https://github.com/NREL/OpenStudio-HPXML/assets/37630588/b8b9ac98-85e0-4794-9921-ab452d9febcd)